vyper/parser/constants.py

```diff
@@ -71,6 +72,28 @@ def unroll_constant(self, const, global_ctx):
             if is_special_case_uint256_literal or is_special_case_int256_literal:
                 fail = False
 
+            # For StructType
+            if is_instances([expr.typ, annotation_type], StructType):
+                for x, y, z in zip(expr.typ.members.keys(), annotation_type.members.keys(), expr.args):
```

Comment:
This may not work in every case.  `expr.typ.members` is a standard `dict` so it's unordered.  So `expr.typ.members.keys()` may not necessarily always return appropriate member names for each value in `expr.args`.  It may happen that it works in this case (since I think the members are added to the dict in the right order and Python happens to have an order-preserving internal dict implementation right now), but I'd rather that we treat an unordered collection as such.

Reply:
Here's another example of how this could go wrong:
```python
CONST_BAR: constant(Foo) = Foo({b: 2, a: 1})
```
This should be possible but this code won't handle that case properly since I don't think `expr.args` will have the same order as `annotation_type.members.keys()`.